packages/client/e2e/analytics.spec.ts

```diff
@@ -0,0 +1,145 @@
+import { expect, Page } from "@playwright/test"
+
+import { test } from "./fixtures"
+import { ElementTimeout, isOpenFin } from "./utils"
+
+const currencyPairs = [
+  "EURUSD",
+  "USDJPY",
+  "GBPUSD",
+  "GBPJPY",
+  "EURJPY",
+  "AUDUSD",
+  "NZDUSD",
+  "EURCAD",
+]
+
+const currencies = ["NZD", "USD", "JPY", "GBP", "EUR", "CAD", "AUD"]
+
+test.describe("Analytics", () => {
+  let analyticsPage: Page
+
+  test.beforeAll(async ({ context, fxPagesRec }, workerInfo) => {
+    if (isOpenFin(workerInfo)) {
+      analyticsPage = fxPagesRec["fx-analytics"]
+    } else {
+      const pages = context.pages()
+      const mainWindow = pages.length > 0 ? pages[0] : await context.newPage()
+
+      await mainWindow.goto(`${process.env.E2E_RTC_WEB_ROOT_URL}`)
+
+      analyticsPage = mainWindow
+    }
+  })
+
+  test.describe("Profit & Loss section", () => {
+    test("Last position amount is displayed in numerical format @smoke", async () => {
+      const lastPositionAmount = await analyticsPage
+        .locator("[data-testid='lastPosition']")
+        .textContent()
+
+      const regexp = RegExp(`[-+,.0-9]`, "g")
+      expect(lastPositionAmount).toMatch(regexp)
+    })
+
+    test("Last position amount is updated periodically", async () => {
+      const lastpositionLocator = analyticsPage.locator(
+        "[data-testid='lastPosition']",
+      )
+      const initiallastPositionAmount = await lastpositionLocator.textContent()
+
+      if (initiallastPositionAmount)
+        await expect(lastpositionLocator).not.toContainText(
+          initiallastPositionAmount,
+          { timeout: ElementTimeout.NORMAL },
+        )
+    })
+
+    test("Correct text color is displayed based on position value", async () => {
+      const initialpositionLocator = analyticsPage.locator(
+        "[data-testid='lastPosition']",
+      )
+      const amount = Number(
+        (await initialpositionLocator.textContent())?.replace(/[.,]/g, ""),
+      )
+
+      if (amount < 0) {
+        expect(await initialpositionLocator.getAttribute("color")).toEqual(
+          "negative",
+        )
+      } else
+        expect(await initialpositionLocator.getAttribute("color")).toEqual(
+          "positive",
+        )
+    })
+
+    test.skip("Graph display coherent values", async () => {
+      // TODO
+      // more complex. may requires datatest-id to reliably assert graph content
+    })
+  })
+
+  test.describe("Positions section", () => {
+    test("Position nodes are showing tooltip information for each currencies", async () => {
+      for (const currency of currencies) {
```

Comment:
minor/thought: I don't think we need to check _all_ of them, do we?

Reply:
Since it is not time costly my thought is it worth checking them all to cover the case where we get data corruption for a single currency